lib/jenkins_pipeline_builder/compiler.rb

```diff
@@ -44,9 +44,15 @@ def get_settings_bag(item_bag, settings_bag = {})
       my_settings_bag.merge(bag)
     end
 
+    def do_compile(item, settings = {})
```

Comment:
I'd really like a better name than do_compile. What does this do in addition to compiling?